docs/Makefile

```diff
@@ -9,15 +9,18 @@ all: create_output
 # where are the example cpp files
 EXAMPLES = $(wildcard examples/*.cpp)
 
+cxx_standard = $(lastword c++11 $(filter c++%, $(subst ., ,$1)))
+
 # create output from a stand-alone example file
 %.output: %.cpp
-	$(MAKE) $(<:.cpp=) CPPFLAGS="-I $(SRCDIR)" CXXFLAGS="-std=c++11"
+	@echo "standard $(call cxx_standard $(<:.cpp=))"
+	$(MAKE) $(<:.cpp=) CPPFLAGS="-I $(SRCDIR)" CXXFLAGS="-std=$(call cxx_standard,$(<:.cpp=)) -Wno-deprecated-declarations"
 	./$(<:.cpp=) > $@
 	rm $(<:.cpp=)
 
 # compare created output with current output of the example files
 %.test: %.cpp
-	$(MAKE) $(<:.cpp=) CPPFLAGS="-I $(SRCDIR)" CXXFLAGS="-std=c++11"
+	$(MAKE) $(<:.cpp=) CPPFLAGS="-I $(SRCDIR)" CXXFLAGS="-std=$(call cxx_standard,$(<:.cpp=)) -Wno-deprecated-declarations"
```

Comment:
I also thought about suppressing deprecation warnings - still, we should make sure we only call deprecation functions in examples for exactly those deprecated functions and not in other situations. I will check after the 3.11.0 release.